lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -34,7 +34,7 @@ def serializable_hash_for_single_resource(options)
       def resource_relationships(options)
         relationships = {}
         serializer.associations(@include_tree).each do |association|
-          relationships[association.key] = relationship_value_for(association, options)
+          relationships[association.key] ||= relationship_value_for(association, options)
```

Comment:
yeah, discovered it in https://github.com/bf4/active_model_serializers/compare/serializer_can_fully_serialize_itself...wip/serializer_can_fully_serialize_itself and was kind of floored by the performance impact